x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -192,6 +193,6 @@ protected static Batch<LogicalPlan> operators() {
     }
 
     protected static Batch<LogicalPlan> cleanup() {
-        return new Batch<>("Clean Up", new ReplaceLimitAndSortAsTopN());
+        return new Batch<>("Clean Up", new ReplaceLimitAndSortAsTopN(), new ReplaceRowAsLocalRelation());
```

Comment:
This is a nice idea, but the issue is that the substitutions happen too early -- all the evaluation of what can be in ROW happens only later. If ROW was a source _only_, we could have given rid of it directly after parsing. But since we allow expressions within it, we need to have `Row` go through "Operator Optimization" batch (so "EVAL" it).